packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -27,80 +27,71 @@ const sanitizeOverFlowingChildren = (spanMap: Map<string, Span>): Map<string, Sp
 
   spanIds.forEach(spanId => {
     const span = spanMap.get(spanId)!;
-    if (span && span.references.length) {
-      // parentSpan will be undefined when its parentSpan is dropped previously
-      const parentSpan = spanMap.get(span.references[0].spanID);
-      if (parentSpan) {
-        const childEndTime = span.startTime + span.duration;
-        const parentEndTime = parentSpan.startTime + parentSpan.duration;
-        switch (true) {
-          case span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime:
-            // case 1: everything looks good
-            // |----parent----|
-            //   |----child--|
-            break;
+    if (!(span && span.references.length)) {
+      return;
+    }
+    // parentSpan will be undefined when its parentSpan is dropped previously
+    const parentSpan = spanMap.get(span.references[0].spanID);
+    if (parentSpan) {
+      const childEndTime = span.startTime + span.duration;
+      const parentEndTime = parentSpan.startTime + parentSpan.duration;
+      if (span.startTime >= parentSpan.startTime) {
+        if (span.startTime >= parentEndTime) {
+          // child outside of parent range => drop the child span
+          //      |----parent----|
+          //                        |----child--|
+          // Remove the childSpan from spanMap
+          spanMap.delete(span.spanID);
 
-          case span.startTime < parentSpan.startTime &&
-            childEndTime <= parentEndTime &&
-            childEndTime > parentSpan.startTime:
-            // case 2: child start before parent, truncate is needed
+          // Remove the childSpanId from its parent span
+          parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id === span.spanID);
+          spanMap.set(parentSpan.spanID, { ...parentSpan });
+        } else if (childEndTime > parentEndTime) {
+            // child end after parent, truncate is needed
             //      |----parent----|
-            //   |----child--|
+            //              |----child--|
             spanMap.set(span.spanID, {
               ...span,
-              startTime: parentSpan.startTime,
-              duration: childEndTime - parentSpan.startTime,
+              duration: parentEndTime - span.startTime,
```

Comment:
this is a serious bug: you are overriding the source data of the trace, literally changing how it looks on the screen, not just decorating it. I think you can easily account for the difference in CP code by using some min/max functions.

Reply:
But we are just working on the copy of the spanMap which is received as parameter to the function sanitizeOverFlowingChildren function and We are also not modifying the original span data.

Reply:
Or instead can we send a copy of the trace data to the memoizedTraceCriticalPath function instead of sending data directly?? So that original data will not be effected

https://github.com/GLVSKiriti/jaeger-ui/blob/9da89cc3633442dabec033dbfec4eed2d8c74c30/packages/jaeger-ui/src/components/TracePage/index.tsx#L394

Here can we send memoizedTraceCriticalPath({...data}) ?? Will the problem solved then??

Reply:
@yurishkuro can you clarify my doubt here!!

Reply:
I think it's fine, I misread it as changing in place, but it's creating a clone of the span, so no issue.